### PR TITLE
Clan Invites Repository Layer

### PR DESCRIPTION
**Pull request by metal-face**

### CODE REVIEW

File: app/repositories/clan_invites.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            clan_invite_id: clan_invite_id,
+            clan_id: clan_id,
+            uses: uses,
+            max_uses: max_uses,
+            expires_at: expires_at,
+            created_at: created_at,
+            updated_at: updated_at,
+        },
+    )
+
```
Comment (cmyui) [inline app/repositories/clan_invites.py]:
```suggestion
    clan_invite = await clients.database.fetch_one(
        query=f"""\
            INSERT INTO clan_invites (clan_invite_id, clan_id, uses, max_uses, expires_at)
            VALUES (:clan_invite_id, :clan_id, :uses, :max_uses, :expires_at)
            RETURNING {READ_PARAMS}
        """,
        values={
            clan_invite_id: clan_invite_id,
            clan_id: clan_id,
            uses: uses,
            max_uses: max_uses,
            expires_at: expires_at,
        },
    )

```

File: app/repositories/clan_invites.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+from app import clients
+from datetime import datetime
+
+
+class ClanInvite(TypedDict):
+    clan_invite_id: UUID
+    clan_id: int
+    uses: int
+    max_uses: int
```
Comment (infernalfire72) [inline app/repositories/clan_invites.py]:
```suggestion
    max_uses: int | None
```

File: app/repositories/clan_invites.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    created_at,
+    updated_at
+"""
+
+
+async def create(
+    clan_invite_id: UUID,
+    clan_id: int,
+    uses: int,
+    max_uses: int,
```
Comment (infernalfire72) [inline app/repositories/clan_invites.py]:
```suggestion
    max_uses: int | None,
```

**Review by infernalfire72 — APPROVED** [review]

File: app/repositories/clan_invites.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    )
+
+    return cast(ClanInvite, clan_invite) if clan_invite is not None else None
+
+
+async def partial_update(
+    clan_invite_id: UUID,
+    clan_id: int | None,
+    uses: int | None,
+    expires_at: datetime | None,
```
Comment (cmyui) [inline app/repositories/clan_invites.py]: nice dx to have defaults
```suggestion
    clan_id: int | None = None,
    uses: int | None = None,
    expires_at: datetime | None = None,
```

File: app/repositories/clan_invites.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        },
+    )
+
+    assert clan_invite is not None
+    return cast(ClanInvite, clan_invite)
+
+
+async def fetch_many(
+    page: int | None,
+    page_size: int | None,
```
Comment (cmyui) [inline app/repositories/clan_invites.py]:
```suggestion
    page: int | None = None,
    page_size: int | None = None,
```

**Review by cmyui — COMMENTED** [review]
+1 👍🏻

**Pull request merged**